utils/yoke/src/crabbake.rs

```diff
@@ -0,0 +1,17 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::Yoke;
+use crate::Yokeable;
+use crabbake::{Bakeable, CrateEnv, TokenStream};
+
+impl<Y, C> Bakeable for Yoke<Y, C>
+where
+    Y: for<'a> Yokeable<'a>,
+    for<'a> <Y as Yokeable<'a>>::Output: Bakeable,
+{
+    fn bake<'a>(&'a self, ctx: &CrateEnv) -> TokenStream {
+        self.get().bake(ctx)
```

Comment:
Question: How does this work? I would assume that you'd need to wrap it in a Yoke constructor

Reply:
You mean wrap the output in a Yoke constructor? I want `bake` to just bake the data struct when I call it on `DataPayload<M>`.

Reply:
(this is analogous to the `Serialize` impl on `Yoke`)

Reply:
This needs to be a `Bakeable` implementation because we make the `Yoke` into a `dyn Bakeable` (`CrabbakeBox`)

Reply:
Unlike with `Serialize`, this `impl Bakeable` is just wrong, though. The only real contract of `Bakeable` is that the token stream it produces must instantiate an instance of the object. This impl fails to satisfy that contract. The contract of `Serialize` is different: it must produce something that is sufficient that `Deserialize` can re-create the object.

Reply:
Do you have an alternative?

Reply:
> Do you have an alternative?

Yes:

1. In crabbake.rs, add a new private trait; let's call it `Foo`
2. Implement `Foo` on `Yoke<Y,C> where Y::Output: Bakeable`
3. Change the `Box<dyn Bakeable>` to a `Box<dyn Foo>`

I think that should work

Reply:
We can also have `yoke` optionally depend on `crabbake` and just have an inherent method. We'd still need the trait though